src/image_export.py

```diff
@@ -0,0 +1,135 @@
+TASK_METADATA = {
+    "display_name": "Artifact Extraction",
+    "description": "Extract artifacts",
+    "task_config": [
+        {
+            "name": "artifacts",
+            "label": "Artfifacts",
+            "description": "A comma seperated list of artifacts to extract",
+            "type": "text",
+            "required": True,
+        },
+    ],
+}
+
+
+@celery.task(bind=True, name=TASK_NAME, metadata=TASK_METADATA)
+def artifact_extract(
+    self,
+    pipe_result: str = None,
+    input_files: list = None,
+    output_path: str = None,
+    workflow_id: str = None,
+    task_config: dict = None,
+) -> str:
+    """Run image_export on input files.
+
+    Args:
+        pipe_result: Base64-encoded result from the previous Celery task, if any.
+        input_files: List of input file dictionaries (unused if pipe_result exists).
+        output_path: Path to the output directory.
+        workflow_id: ID of the workflow.
+        task_config: User configuration for the task.
+
+    Returns:
+        Base64-encoded dictionary containing task results.
+    """
+    input_files = get_input_files(pipe_result, input_files or [])
+    output_files = []
+    artifacts = task_config["artifacts"].split(",")
+    for artifact in artifacts:
+        for input_file in input_files:
+            log_file = create_output_file(
+                output_path, filename=f"image_export_{artifact}", file_extension="log"
+            )
+
+            export_directory = os.path.join(output_path, uuid4().hex)
+            os.mkdir(export_directory)
+
+            command = [
+                "image_export.py",
+                "--no-hashes",
+                "--logfile",
+                log_file.path,
+                "--write",
+                export_directory,
+                "--partitions",
+                "all",
+                "--volumes",
+                "all",
+                "--unattended",
+                "--artifact_filters",
+                artifact,
+                input_file.get("path"),
+            ]
+            command_string = " ".join(command[:5])
+
+            process = subprocess.Popen(command)
+            while process.poll() is None:
+                if os.path.isfile(log_file.path):
+                    with open(log_file.path, "r", encoding="utf-8") as f:
+                        log_dict = f.read()
+                        self.send_event("task-progress", data=log_dict)
+                time.sleep(2)
+
+        if os.path.isfile(log_file.path):
+            output_files.append(log_file.to_dict())
+
+        export_directory_path = Path(export_directory)
+        extracted_files = [f for f in export_directory_path.glob("**/*") if f.is_file()]
+        for file in extracted_files:
+            original_path = str(file.relative_to(export_directory_path))
+
+            output_file = create_output_file(
+                output_path=output_path,
+                filename=original_path,
```

Comment:
Should this be the full path? I would expect this to be the original filename of the extracted file. The original full path will be part of the detailed table.

Reply:
I added the filename as a full path as we do not show orignal_path in the table yet. Will set filename to filename only.